File: ApiSuvesaPos/Negocio/Logica/PreTomaProveedor.vb
Namespace Logica
    Public Class PreTomaProveedor

        Private db As Datos.Class.PreTomaProveedor

        Sub New()
            Me.db = New Datos.Class.PreTomaProveedor
        End Sub

        Public Function Crear(pretoma As Datos.Models.PreTomaProveedor) As String
            Return Me.db.Crear(pretoma)
        End Function

        Public Function Editar(id As Decimal, pretoma As Datos.Models.PreTomaProveedor) As String
            Dim res As String = Me.db.Editar(id, pretoma)
            If res = "0" Then
                Return "No existe el valor"
            Else
                Return res
            End If
        End Function

        Public Function Buscar(porNombre As Boolean, Filtro As String) As List(Of Datos.Models.PreTomaProveedor)
            Dim datos As New List(Of Datos.Models.PreTomaProveedor)
            datos = db.Buscar(porNombre, Filtro)

            Dim Resultado As New List(Of Datos.Models.PreTomaProveedor)
            For Each toma As Datos.Models.PreTomaProveedor In datos
                Dim preToma As New Datos.Models.PreTomaProveedor
                preToma.Id = toma.Id
                preToma.IdUsuarioCreo = toma.IdUsuarioCreo
                preToma.CodigoProv = toma.CodigoProv
                preToma.Fecha = toma.Fecha
                preToma.Anulado = toma.Anulado
                preToma.Usado = toma.Usado

                For Each detalle As Datos.Models.PreTomaProveedorDetalle In Me.db.BuscarDetalle(preToma.Id)
                    Dim preTomaDetalle As New Datos.Models.PreTomaProveedorDetalle
                    preTomaDetalle.Id = detalle.Id
                    preTomaDetalle.IdPreTomaProveedor = detalle.IdPreTomaProveedor
                    preTomaDetalle.Codigo = detalle.Codigo
                    preTomaDetalle.CodArticulo = detalle.CodArticulo
                    preTomaDetalle.Descripicon = detalle.Descripicon
                    preTomaDetalle.Cantidad = detalle.Cantidad

                    preToma.PreTomaProveedorDetalles.Add(preTomaDetalle)
                Next

                Resultado.Add(preToma)
            Next

            Return Resultado
        End Function

    End Class
End Namespace

File: ApiSuvesaPos/Datos/Class/PreTomaProveedor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Datos.Models;
using Microsoft.EntityFrameworkCore;

namespace Datos.Class
{
	public class PreTomaProveedor
	{
		private SeePOSContext entities;

		public PreTomaProveedor()
		{
			entities = new SeePOSContext();
		}

		public int Crear(Models.PreTomaProveedor pretomaproveedor) // registro de PreTomaProveedor
		{
			try
			{
				entities.PreTomaProveedors.Add(pretomaproveedor);
				return entities.SaveChanges();
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}

		public List<Models.PreTomaProveedorDetalle> BuscarDetalle(decimal Id)  //consultar PreTomaProveedor
		{
			try
			{
				List<Models.PreTomaProveedorDetalle> result;
					var temp = from c in entities.PreTomaProveedorDetalles
							   where c.IdPreTomaProveedor == Id
							   select c;
					result = temp.ToList<Models.PreTomaProveedorDetalle>();
				
				if (result.Count > 0)
				{
					return result;
				}
				else
				{
					return result = null;
				}
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}

		public List<Models.PreTomaProveedor> Buscar(bool porNombre, string filtro)  //consultar PreTomaProveedor
		{
			try
			{
				List<Models.PreTomaProveedor> result;
				if (porNombre == true)
				{
					var temp = from c in entities.PreTomaProveedors
							   select c;
					result = temp.ToList<Models.PreTomaProveedor>();
				}
				else
				{
					var temp = from c in entities.PreTomaProveedors
							   select c;
					result = temp.ToList<Models.PreTomaProveedor>();
				}
				if (result.Count > 0)
				{
					return result;
				}
				else
				{
					return result = null;
				}
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}

		public int Editar(decimal id, Models.PreTomaProveedor nuevo) //editar PreTomaProveedor
		{
			try
			{
				var p = entities.PreTomaProveedors.Find(id);
				Models.PreTomaProveedor viejo = p;
				if (viejo != null)
				{
					viejo.Id = nuevo.Id;
					viejo.IdUsuarioCreo = nuevo.IdUsuarioCreo;
					viejo.CodigoProv = nuevo.CodigoProv;
					viejo.Fecha = nuevo.Fecha;
					viejo.Anulado = nuevo.Anulado;
					viejo.Usado = nuevo.Usado;

					entities.Entry(viejo).State = EntityState.Modified;

					return entities.SaveChanges();
				}
				else
				{
					return 0;// no se encotro el registro solicitado.
				}
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}

		public int Borrar(decimal id) //borrar PreTomaProveedor
		{
			try
			{
				var p = entities.PreTomaProveedors.Find(id);
				entities.Remove(p);
				return entities.SaveChanges();
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}
	}
}


